### PR TITLE
Fix a bug due to variable scoping in multi-threaded environment. Also…

### PR DESCRIPTION
**Pull request by Tyf0x**
… fix an error leading to a wrong initialisation of the system.

### CODE REVIEW

File: hashids.cfc [hunk 1]
```diff
@@ -159,7 +160,7 @@ component output="false" persistent="false" {
 		var sepDiff = alphabetLen / sepsLen;
 
 		if( !len(seps) || (sepDiff > getSepDiv()) ){
-			sepsLen = ceiling(sepDiff);
+			sepsLen = ceiling(alphabetLen / getSepDiv());
```
Comment (dswitzer) [inline hashids.cfc:163]: This change breaks the test_specifying_custom_hash_alphabet. This test is to make sure the output is compatible with the Java version of the implementation:

https://github.com/jiecao-fm/hashids-java/blob/master/src/test/java/fm/jiecao/lib/HashidsTest.java

**Comment by Tyf0x** [comment]
I believe that the tests are outdated and incorrect.

I've downloaded the latest version of the [java Hashid source](https://github.com/jiecao-fm/hashids-java/archive/hashids-1.0.1.zip) and added a test as follow:

``` java
    @Test
    public void test_specifying_custom_hash_alphabet(){
        String expected = "b332db5", res;
        long num_to_hash = 1234567L;
        long[] res2;
        Hashids a = new Hashids("this is my salt", 0, "0123456789abcdef");
        res = a.encode(num_to_hash);
        Assert.assertEquals(expected, res);
        res2 = a.decode(expected);
        Assert.assertEquals(res2[0], num_to_hash);
    }
```

The test is a success:
shot 2016-02-04

This confirms that encoding 1234567 should hash it to "b332db5" and not "bbdd589a".

Edit: You can double-check with the [JSversion](http://codepen.io/anon/pen/VeGMgo).

Does this make my patch acceptable?

**Comment by dswitzer** [comment]
@Tyf0x:

Thanks for the proof. If you update the unit test with the fixed version, then I'll definitely accept the pull request. That was the only issue I spotted.

Thanks for taking the time to put this together.

**Comment by Tyf0x** [comment]
@dswitzer :
No worries at all, glad it helps ;)

**Comment by Tyf0x** [comment]
Any chance of this being merged? (I updated the test case)

Thanks.

**Pull request merged**